transport/rif_comms/client.py

```diff
@@ -8,95 +8,101 @@
-        return self.stub.CreateTopicWithRskAddress(RskAddress(address=topic_id))
+        # TODO: catch already subscribed and any error
+        return self.stub.CreateTopicWithRskAddress(RskAddress(address=rsk_address))
 
-    # TODO review params and create docstring. It has no sense to use both peer id and rsk_address
-    def has_subscription(self, rsk_address: Address) -> BooleanResponse:
-        peer_id = self.get_peer_id(rsk_address)
+    def is_subscribed_to(self, rsk_address: Address) -> bool:
+        """
+        Returns whether or not the client's underlying RIF Communications node is subscribed to the topic
+        which corresponds to the given RSK address.
+        Invokes HasSubscriber GRPC API endpoint.
+        :param rsk_address: RSK address which corresponds to the topic which is being checked for subscription
+        :return: boolean value indicating whether the client is subscribed or not
+        """
+        our_peer_id = self._get_peer_id(self.rsk_address)
+        topic_id = self._get_peer_id(rsk_address)
         return self.stub.HasSubscriber(
             Subscriber(
-                peerId=peer_id,
-                channel=Channel(channelId=rsk_address)
+                peerId=our_peer_id,
+                channel=Channel(channelId=topic_id)
             )
-        )
+        ).value
 
-    def send_message(self, topic_id: str, data: str) -> Void:
+    def send_message(self, payload: str, rsk_address: Address):
         """
-        Sends a message to a topic.
-        Invokes the SendMessageToTopic grpc api endpoint
-        :param topic_id: topic identifier
-        :param message: the message data
-        :return: void
+        Sends a message to a destination RSK address.
+        Invokes the SendMessageToTopic GRPC API endpoint.
+        :param payload: the message data to be sent
+        :param rsk_address: the destination for the message to be sent to
         """
-
-        # TODO message encoding
-        return self.stub.SendMessageToTopic(
+        topic_id = self._get_peer_id(rsk_address)
+        # TODO: message encoding
+        self.stub.SendMessageToTopic(
             PublishPayload(
                 topic=Channel(channelId=topic_id),
-                message=Msg(payload=str.encode("Test message"))
+                message=Msg(payload=str.encode(payload))
             )
         )
 
-    def unsubscribe(self, topic_id: str) -> Void:
+    def unsubscribe_from(self, rsk_address: Address):
         """
-        This unsubscribe the node from the topic.
-        Invokes the CloseTopic grpc api endpoint.
-        :param topic_id: topic identifier
-        :return: void
+        Unsubscribes from a topic which corresponds to the given RSK address.
+        Invokes the CloseTopic GRPC API endpoint.
+        :param rsk_address: RSK address which corresponds to the topic which the client is unsubscribing from.
         """
+        topic_id = self._get_peer_id(rsk_address)
```

Comment:
same

Reply:
resolving this thread so we keep the discussion in one place